### ISSUE TITLE
Publisher leaks a batch-actor task and unbounded channel per ordering key

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#6723]
Unconfirmed.  This is the result of a search with Claude, may be a false positive.

In `src/pubsub/src/publisher/actor.rs`, publishing with an ordering key spawns a dedicated `BatchActor` task and adds an entry to `batch_actors`:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/pubsub/src/publisher/actor.rs#L130-L147

```rust
// TODO(#4012): Remove batch actors when there are no outstanding operations on the ordering key.
```

Batch actors and their associated task handles and channels are never removed from `batch_actors`. In applications that use dynamic or short-lived ordering keys (e.g. per-entity or per-session keys), this creates an unbounded memory and background task leak.

**Comment by suzmue** [comment]
This is a known issue with the publisher + ordering keys. Solving this would also require running some benchmarking to investigate what the effect of removing and rapidly recreating these batch actors would be to determine when is the best time to clean up (immediately when its empty? after a certain number of ticks of batch delay? after a certain number of messages have been received with none in the ordering key?)

**Comment by coryan** [comment]
Should we close this as a dup?

**Comment by suzmue** [comment]
Looks like the TODO references a stale issue, I think we can leave this open to track this work.
